sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/EventStreamListener.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+/**
+ * Interface to implement event stream listeners for handling {@link ServerSentEvent}.
+ */
+public interface EventStreamListener {
+
+    /**
+     * Gets called every time an event or data is received.
+     *
+     * @param sse the instance of {@link ServerSentEvent}
+     */
+    void onEvent(ServerSentEvent sse);
+
+    /**
+     * Gets called if an error has occurred
+     *
+     * @param throwable Error that occurred
+     */
+    void onError(Throwable throwable);
```

Comment:
We should make these default methods and only `onEvent` should really be required to be implemented making this a functional interface.